sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Window.java

```diff
@@ -486,7 +486,7 @@ public void populateDisplayData(DisplayData.Builder builder) {
           // We first apply a (trivial) transform to the input PCollection to produce a new
           // PCollection. This ensures that we don't modify the windowing strategy of the input
           // which may be used elsewhere.
-          .apply("Identity", MapElements.via(new SimpleFunction<T, T>() {
+          .apply("Identity", MapElements.<T,T>via(new SimpleFunction<T, T>() {
```

Comment:
@davorbonaci 

I realized that my PR description was also ambiguous -- IIRC, the problem wasn't that compiler couldn't determine T, but rather the ambiguity was between SimpleFunction and SerializableFunction -- I think it was saying that it didn't know which overloaded copy of "via" to call due to type erasure  (in case that changes the guidance you'd want to give about this)